### ISSUE TITLE
Wondrous Lands Problems with Hero Selection and Achievements Page

### ISSUE THREAD

**Issue by Esortsyt** [ruffle-rs/ruffle#14607]
### Describe the bug

The Hero Selection does this every time I go in
![Screenshot 2024-01-03 5 09 23 PM](https://github.com/ruffle-rs/ruffle/assets/155596413/d0c03d6b-b784-4eb9-9b4f-b155f368fb27)

And the Achievements page loads these achievement images super fast and changes the text to russian
![Screenshot 2024-01-03 5 14 09 PM](https://github.com/ruffle-rs/ruffle/assets/155596413/41ffc9c2-1bf3-43ec-87c6-209883f98aab)
![Screenshot 2024-01-03 5 14 13 PM](https://github.com/ruffle-rs/ruffle/assets/155596413/0e86ee2f-0a2d-4970-8ada-621fba6e113b)
![Screenshot 2024-01-03 5 14 17 PM](https://github.com/ruffle-rs/ruffle/assets/155596413/da58d00e-8109-439c-a864-ee86ff20c105)
![Screenshot 2024-01-03 5 14 21 PM](https://github.com/ruffle-rs/ruffle/assets/155596413/4724713a-9bd2-4e17-aaf3-987bf23e7f80)
![Screenshot 2024-01-03 5 15 46 PM](https://github.com/ruffle-rs/ruffle/assets/155596413/025c59b3-84f8-428a-8ddc-1ef9911adcbf)
![Screenshot 2024-01-03 5 16 17 PM](https://github.com/ruffle-rs/ruffle/assets/155596413/3d9e727d-2cb9-41a1-bafd-d53c554837b5)
![Screenshot 2024-01-03 5 16 20 PM](https://github.com/ruffle-rs/ruffle/assets/155596413/84933391-f044-4376-9719-52497868d4dc)

### Expected behavior

Should allow me to pick heroes and display the stats of each hero correctly without breaking. It should also display each achievement and image in each box and not one achievement image in all boxes, cycling through all images super fast. It should also not turn the text "Achievements" russian, revert to english and then repeat.

### Content Location

https://ooooooooo.ooo/static/?29d4d783-91f6-449e-8e84-c6eacf99af40

^ leads to Wondrous Lands in Flashpoint Archive

### Affected platform

Self-hosted version

### Operating system

Chrome OS

### Browser

Google Chrome

**Comment by n0samu** [comment]
The SWF can be accessed from this URL: https://cache.armorgames.com/files/games/wondrous-lands-18251.swf
Relevant errors:
```
ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "rollOut", class: flash.events::MouseEvent, ptr: 0x1fe619b1050 }) to handler FunctionObject(FunctionObject { ptr: 0x1fe6941b300, name: Ok("screens.heroinfoscreen::TextButtonClass/RollOut()") }) : TypeError: Error #1009: Cannot access a property or method of a null object reference. (accessing field: text)
        at screens.heroinfoscreen::TextButtonClass/UpdateText()
        at screens.heroinfoscreen::TextButtonClass/RollOut()
ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "mouseUp", class: flash.events::MouseEvent, ptr: 0x1fe696c7a50 }) to handler FunctionObject(FunctionObject { ptr: 0x1fe653213c0, name: Ok("mainmenu::IntroMainScreen/NextButtonMouseUp()") }) : Error: Нет такого типа урона
        at text::EquipController$/ConvertWeaponClass()
        at equip::EquipStorage$/GetEquipStruct()
        at equip::EquipStorage$/InitializeEquipment()
        at equip::EquipStorage$cinit()
        at global$init() [TU=]
        at equip::RandomEquipGenerator$/GenerateWeapon()
        at equip::RandomEquipGenerator$/GenerateEquip()
        at equip::EquipEcnhantHelper$/GenerateEquipWithLevelStruct()
        at equip::EquipGenerator$/GetNormalEquip()
        at equip::EquipGenerator$/GenerateEquipByLevel()
        at utilities::HeroEquipGenerator$/GenerateEquipInternal()
        at utilities::HeroEquipGenerator$/GenerateHeroByStructs()
        at utilities::HeroEquipGenerator$/GenerateWarrierEquip()
        at utilities::HeroEquipGenerator$/GenerateEquip()
        at utilities::HeroParamWrapper/GenerateEquip()
        at utilities::HeroGenerator$/GenerateHero()
        at screens.herochoicescreen::HeroChoiseScreen/GenerateHeroes()
        at screens.herochoicescreen::HeroChoiseScreen/InitializeHeroChoise()
        at mainmenu::MenuScreen/ShowHeroChoise()
        at mainmenu::IntroMainScreen/View()
        at mainmenu::IntroMainScreen/NextButtonMouseUp()
```

**Comment by torokati44** [comment]
Was this also fixed by https://github.com/ruffle-rs/ruffle/pull/14764 ?

**Comment by Esortsyt** [comment]
Idk

**Comment by n0samu** [comment]
Fixed by #14689.
